src/NuGet.Clients/NuGet.PackageManagement.UI/Automation/TabItemButtonAutomationPeer.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Windows.Automation.Peers;
+using System.Windows.Automation.Provider;
+
+namespace NuGet.PackageManagement.UI
+{
+    /// <summary>
+    /// Automation peer to Represents Button controls which are used as Clickable Tab Items.
+    /// This changes the ControlType for such buttons to TabItems and provides the necessary
+    /// AutomationPatterns suitable for a tab item. 
+    /// Example: Browse, Installed, Update Tab items in the Nuget Package Manager tab
+    /// </summary>
+    internal class TabItemButtonAutomationPeer : FrameworkElementAutomationPeer, ISelectionItemProvider
+    {
+        private TabItemButton _tabItemButton;
+
+        public TabItemButtonAutomationPeer(TabItemButton owner) : base(owner)
+        {
+            Debug.Assert(owner != null);
```

Comment:
This should throw `ArgumentNullException`. As an alternative you can use `Assumes.NotNull`